apps/accounts/validators.py

```diff
@@ -0,0 +1,129 @@
+import datetime
+from django.core.exceptions import ValidationError
+from django.db.models import Q
+from django_password_validators.password_history.models import (
+    PasswordHistory,
+    UserPasswordHistoryConfig,
+)
+from django_password_validators.password_history.password_validation import UniquePasswordsValidator
+
+
+class UniqueAndMinAgedPasswordsValidator(UniquePasswordsValidator):
+
+    def __init__(self,
+                 password_min_age=60 * 60 * 24,
+                 password_reuse_interval=60 * 60 * 24 * 120,
+                 password_expire=60 * 60 * 24 * 30):
+
+        msg1 = "Invalid OPTIONS, password_min_age < password_reuse_interval expected, " \
+               "but having password_min_age({}) < password_reuse_interval({})"
+        msg2 = "Invalid OPTIONS, password_expire < password_reuse_interval expected, " \
+               "but having password_expire({}) < password_reuse_interval({})"
+        msg3 = "Invalid OPTIONS, password_min_age < password_expire expected, " \
+               "but having password_expire({}) < password_reuse_interval({})"
+
+        if password_min_age > 0 and password_reuse_interval > 0:
+            assert password_min_age <= password_reuse_interval, \
+                msg1.format(password_min_age, password_reuse_interval)
+        if password_expire > 0 and password_reuse_interval > 0:
+            assert password_expire <= password_reuse_interval, \
+                msg2.format(password_expire, password_reuse_interval)
+        if password_min_age > 0 and password_expire > 0:
+            assert password_min_age <= password_expire, \
+                msg3.format(password_min_age, password_expire)
```

Comment:
I'm assuming we may not want to use assert here since we just cleaned up asserts in a previous PR and it prevents us from turning on python optimization in the future?

Reply:
right, they can be replaced by ValueError, I am think about how to get these settings validated at bb2 start up time.